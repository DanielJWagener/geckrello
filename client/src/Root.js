// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
