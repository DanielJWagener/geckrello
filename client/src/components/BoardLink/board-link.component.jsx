import React from "react";
import { Link } from "react-router-dom";

import colorThemes from "../../utilities/colorThemes";

import "./board-link.styles.scss";

const BoardLink = props => {
  const backgroundColor = colorThemes[props.background].dark;
  console.log(backgroundColor);

  return (
    <div className="board-link" style={{ backgroundColor }}>
      <Link className="board-link__link" to={`/board/${props.boardId}`}>
        <h1>{props.title}</h1>
      </Link>
    </div>
  );
};

export default BoardLink;
