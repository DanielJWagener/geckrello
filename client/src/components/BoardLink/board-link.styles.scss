.board-link {
  background-color: #444;
  padding: 1rem 2rem;
  border-radius: 0.5rem;
  text-align: center;
  margin-bottom: 1rem;
  font-size: 2rem;
  overflow-x: hidden;
  text-decoration: none;
  cursor: pointer;

  &__link {
    color: white;
    text-decoration: none;

    &:visited {
      color: white;
      text-decoration: none;
    }
  }

  &:hover > .board-link__link {
    color: #ddd;
  }
}
