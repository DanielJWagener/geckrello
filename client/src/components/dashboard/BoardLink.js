import React from "react";
import { Link } from "react-router-dom";

const BoardLink = props => {
  return (
    <Link to={`/board/${props.boardId}`}>
      <h1>{`I'm a link to ${props.title}`}</h1>;
    </Link>
  );
};

export default BoardLink;
