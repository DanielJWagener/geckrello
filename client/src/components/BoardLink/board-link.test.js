import { shallow } from "enzyme";
import React from "react";
import BoardLink from "./board-link.component";
import Root from "../../Root";

it("renders BoardLink component", () => {
  expect(shallow(<BoardLink background="blue" />).debug()).toMatchSnapshot();
});
