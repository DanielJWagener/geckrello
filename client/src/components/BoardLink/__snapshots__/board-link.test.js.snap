// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BoardLink component 1`] = `
"<Link className=\\"board-link\\" to=\\"/board/undefined\\" style={{...}}>
  <div className=\\"board-link__link\\" />
</Link>"
`;
